cermine-impl/src/main/java/pl/edu/icm/cermine/ContentExtractor.java

```diff
@@ -847,20 +844,122 @@ public static void main(String[] args) throws ParseException, AnalysisException,
+                    + "                         for description of available configuration properties\n"
+                    );
+            System.exit(1);
+        }
+
+        final boolean override = parser.override();
+        final Long timeoutSeconds = parser.getTimeout();
+        
+        String path = parser.getPath();
+        final String outpath = parser.getOutPath()+'/';
+        final Map<String, String> extensions = parser.getTypesAndExtensions();
+
+        Long fileLength = Files.list(Paths.get(path)).count();
+        Long getlimit = parser.getLimit();
+        Long workers = parser.getWorkers();
+
+        if (getlimit == null || getlimit > fileLength) {
+            getlimit = fileLength;
+        }
+
+        if (workers > getlimit) {
+            workers = getlimit;
+        }
+
+        ExtractionConfigBuilder builder = new ExtractionConfigBuilder();
+        if (parser.getConfigurationPath() != null) {
+            builder.addConfiguration(parser.getConfigurationPath());
         }
+
+        final Long limit = getlimit;
+        builder.setProperty(ExtractionConfigProperty.IMAGES_EXTRACTION, extensions.containsKey("images"));
+        ExtractionConfigRegister.set(builder.buildConfiguration());
+        final ExecutorService executor = Executors.newFixedThreadPool(workers.intValue());
+
+        final Path startFilePath = Paths.get(new URI("file://"+path));
+
+        long start = System.currentTimeMillis();
+        float elapsed;
+
+        Files.walkFileTree(startFilePath, new SimpleFileVisitor<Path>() {
+            int counter = 0;
+            @Override
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                if (file.toString().toLowerCase().endsWith(".pdf") && counter < limit) {
+                    Runnable worker = new parallelTask(file.toFile(), extensions, override, timeoutSeconds, outpath, limit);
+                    executor.execute(worker);
+                    counter++;
+                } else if (counter >= limit) {
+                    return FileVisitResult.SKIP_SIBLINGS;
+                }
+                return FileVisitResult.CONTINUE;
+            }
+            @Override
+            public FileVisitResult postVisitDirectory(Path dir, IOException e) throws IOException {
+                if (e == null) {
+                    if (dir == startFilePath) {
+                        executor.shutdown();
+                        while (!executor.isTerminated()) {   }
+                    }
+                    return FileVisitResult.CONTINUE;
+                } else {
+                    // directory iteration failed
+                    executor.shutdown();
+                    throw e;
+                }
+            }
+        });
+
+        long end = System.currentTimeMillis();
+        elapsed = (end - start) / 1000F;
+        System.out.println("Total extraction time: " + Math.round(elapsed) + "s");
```

Comment:
The original code was printing the extraction time for every file. This could be added to the `ParallelTask#run()` method. Also "File processed:" message from the original code is missing.